### PR TITLE
My Jetpack: Add new `Text` component at `js-packages/components`

### PR DESCRIPTION
**Pull request by renatoagds**
It creates a new component that will keep consistency across our texts, following all tokens proposed by Design at: PbJqbW-8t-p2

Usage example drafted on #23311 

#### Changes proposed in this Pull Request:

* Creates new `Text` component

#### Jetpack product discussion

PbJqbW-8t-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Start `Storybook`.
* Goes to `Components / Text`.
* Interact with `Controls` tab, you should see all variants.

#### Demo

Screen Shot 2022-03-14 at 18 52 23

### CODE REVIEW

**Review by kraftbj — DISMISSED** [review]
It looks good and makes sense to me. @Automattic/jetpack-agora please confirm and merge when acceptable to y'all.

File: projects/js-packages/components/components/text/index.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+Text.propTypes = {
+	variant: PropTypes.oneOf( Object.keys( VARIANTS_MAPPING ) ),
+	children: PropTypes.node,
+	component: PropTypes.oneOfType( [ PropTypes.elementType, PropTypes.string ] ),
+};
+
+Text.defaultProps = {
+	variant: 'body',
```
Comment (leogermani) [inline projects/js-packages/components/components/text/index.jsx:42]: Can you add doc block with the description of each argument?

File: projects/js-packages/components/components/text/index.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	children: null,
+	component: null,
```
Comment (leogermani) [inline projects/js-packages/components/components/text/index.jsx:44]: question. Is it possible to use anything to test children and component in storybook?

Comment (leogermani) [inline projects/js-packages/components/components/text/index.jsx:44]: question. Is it possible to use anything to test children and component in storybook?

Comment (renatoagds) [reply projects/js-packages/components/components/text/index.jsx:44]: Yes! And I'm already doing that here: https://github.com/Automattic/jetpack/blob/a3fc2db7f758522427a88cd94b2eb20a1e76045e/projects/js-packages/components/components/text/stories/index.jsx#L24-L40

File: projects/js-packages/components/components/text/index.jsx (hunk 1 above)
Comment (renatoagds) [reply projects/js-packages/components/components/text/index.jsx:45]: Done https://github.com/Automattic/jetpack/pull/23309/commits/f4ffdfd331da4f582d902fede89ab400eba5afc1

**Review by leogermani — APPROVED** [review]
Looks great

**Pull request merged**